core/domain/event_services.py

```diff
@@ -40,18 +49,24 @@ class BaseEventHandler:
 
     # A string denoting the type of the event. Should be specified by
     # subclasses and considered immutable.
-    EVENT_TYPE = None
+    EVENT_TYPE: Optional[str] = None
 
+    # Here both the arguments are annotated with Any type because in child
+    # classes this method can be redefined with any number of named and keyword
+    # arguments with different kind of types.
     @classmethod
-    def _handle_event(cls, *args, **kwargs):
+    def _handle_event(cls, *args: Any, **kwargs: Any) -> None:
```

Comment:
I think I'm fine with it. Another option would be to remove `_handle_event` from here and only verify that it exists in the `record()` below, that way we can check if it exists while keeping the signatures up to the child classes.

Reply:
Just to confirm, the second approach that you mentioned is something like:
```python
    @classmethod
    def record(cls, *args: Any, **kwargs: Any) -> None:
           
         if cls._handle_event do not exist:
              raise exception ....

         cls._handle_event(*args, **kwargs)
```

Reply:
I think the approach that @vojtechjelinek mentioned would let us remove the ignore-pragma in `_handle_event: Callable[..., None]  # type: ignore[no-redef]`

@sahiljoster32, you might need to use `getattr(cls, '_handle_event')` instead of `cls._handle_event` in `record()`

Reply:
@sahiljoster32 Yep, smth like that.

Reply:
Overloads are removed!